src/components/atoms/SpacesDropdown/SpacesDropdown.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { Dropdown as ReactBootstrapDropdown } from "react-bootstrap";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+
+import { ADMIN_V1_ROOMS_URL } from "settings";
+
+import { Room } from "types/rooms";
+import { ContainerClassName } from "types/utility";
+
+import { venueSpacesList } from "utils/room";
+
+import { Dropdown } from "components/atoms/Dropdown";
+
+import "./SpacesDropdown.scss";
+
+export interface SpacesDropdownProps extends ContainerClassName {
+  defaultSpace?: string;
+  venueSpaces: Room[];
+  venueId?: string;
+  setValue: <T>(prop: string, value: T, validate: boolean) => void;
+  fieldName: string;
+}
+
+export const SpacesDropdown: React.FC<SpacesDropdownProps> = ({
+  defaultSpace,
+  venueSpaces,
+  venueId,
+  containerClassName,
+  setValue,
+  fieldName,
+}) => {
+  const [spaceValue, setSpaceValue] = useState<string | undefined>(
+    defaultSpace
+  );
+
+  const getSpaceIcon = useCallback(
+    (spaceTemplate) =>
+      venueSpacesList.find(
+        (venueSpace) => venueSpace.template === spaceTemplate
+      )?.icon,
+    []
+  );
+
+  const spaceOptions = useMemo(() => {
+    const options = venueSpaces.map((space) => {
+      const spaceIcon = getSpaceIcon(space.template);
+      return (
+        <ReactBootstrapDropdown.Item
```

Comment:
If we're building reusable dropdown for everything (admin related), should it be based on Bootstrap at all? There's baggage that comes with it which if possible to avoid, would be nice.

Reply:
There is a restriction from `react-bootstrap` to use only `Dropdown.Item`. Can't find the way to avoid it

Reply:
My guess, we avoid by not using Bootstrap elements as basis in some later PR. Good enough for now.